lib/goals.js

```diff
@@ -282,6 +333,10 @@ function distanceXZ (dx, dz) {
   return Math.abs(dx - dz) + Math.min(dx, dz) * Math.SQRT2
 }
 
+function distance (dx, dy, dz) {
+  return Math.sqrt(dx ^ 2 + dy ^ 2 + dz ^ 2)
```

Comment:
I'm pretty sure this is already defined in vec3 in one form or an other.

Reply:
I changed it to Vec3.distanceTo. We could also replace distanceXZ used in Goals a lot with Vec3.xyDistanceTo